sos/policies/runtimes/lxd.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (C) 2023 Canonical Ltd., Arif Ali <arif.ali@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+
+from sos.policies.runtimes import ContainerRuntime
+from sos.utilities import sos_get_command_output
+from sos.utilities import is_executable
+
+
+class LxdContainerRuntime(ContainerRuntime):
+    """Runtime class to use for systems running LXD"""
+
+    name = 'lxd'
+    binary = 'lxc'
+
+    def check_is_active(self, sysroot=None):
+        # the daemon must be running
+        if (is_executable('lxc', sysroot) and
+                (self.policy.init_system.is_running('lxd') or
+                 self.policy.init_system.is_running('snap.lxd.daemon'))):
+            self.active = True
+            return True
+        return False
+
+    def get_containers(self, get_all=False):
+        """Get a list of containers present on the system.
+
+        :param get_all: If set, include stopped containers as well
+        :type get_all: ``bool``
+        """
+        containers = []
+
+        _cmd = "%s list --format json" % (self.binary)
```

Comment:
Nitpick - the rest of the runtime uses f-strings (yay!), so let's use one here to for consistency, and for getting away from string substitution.

Reply:
nice catch, I originally wrote this 5 months ago, so converted all to f-string, missed this one ;)